tensorflow/core/kernels/cuda_sparse.h

```diff
@@ -16,15 +16,38 @@ limitations under the License.
 #ifndef TENSORFLOW_CORE_KERNELS_CUDA_SPARSE_H_
 #define TENSORFLOW_CORE_KERNELS_CUDA_SPARSE_H_
 
-// This header declares the class CudaSparse, which contains wrappers of
+// This header declares the class GpuSparse, which contains wrappers of
 // cuSparse libraries for use in TensorFlow kernels.
 
-#ifdef GOOGLE_CUDA
+#if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
 #include <functional>
 #include <vector>
 
+#if GOOGLE_CUDA
```

Comment:
Trying to think from the perspective of the upstream reviewer. Now that you are able to play the trick here to separate compile time dependency between rocm and cuda, why do you need two C++ implementations? Correct me if I'm wrong, my impression in looking at the difference of the two cpp files is that in terms of functionality`rocm_sparse` is a subset of `cuda_sparse`.

Reply:
`rocm_sparse.cc` makes calls to the `hipSPARSE` library, `cuda_sparse.cc` makes calls to the `cuSPARSE` library. That is the primary difference between the two. 

We can play some `using gpu... = ` tricks, to create a single `gpu_sparse.cc` but that would require that `hipSPARSE` support always be in sync with `cuSPARSE`, which is not the case now. Also I do not think it will ever be case. Given that, what this change tries to achieve is keep the same interface for `hipSPARSE` / `cuSPARSE` wrapper (i.e. a common `cuda_sparse.h` file which will get renamed to `gpu_sparse.h`), but have cuda / rocm specific implementation files.

Reply:
That sounds reasonable, thanks for the explanation